wgpu-core/src/instance.rs

```diff
@@ -666,7 +666,7 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
                 log::warn!("max_bind_groups limit is missing");
             } else {
                 assert!(
-                    u32::from(limits.max_bound_descriptor_sets) >= desc.limits.max_bind_groups,
+                    u32::from(adapter.limits.max_bind_groups) >= desc.limits.max_bind_groups,
```

Comment:
these should use the same units, so `u32::from` is not needed
also, let's change the order to `desc.limits.max_bind_groups <= adapter.limits.max_bind_groups` as I think it's more clear